html/modules/custom/reliefweb_import/src/Plugin/ReliefWebImporter/WfpLogClusterImporter.php

```diff
@@ -0,0 +1,639 @@
+          '@id' => $id,
+        ]));
+
+        continue;
+      }
+
+      // Mandatory information.
+      $data['provider'] = $provider_uuid;
+      $data['bundle'] = $bundle;
+      $data['hash'] = $hash;
+      $data['uuid'] = $uuid;
+      $data['url'] = $url;
+
+      // Submit the document directly, no need to go through the queue.
+      try {
+        $entity = $plugin->process($data);
+        $import_record['status'] = 'success';
+        $import_record['message'] = '';
+        $import_record['attempts'] = 0;
+        $import_record['entity_id'] = $entity->id();
+        $import_record['entity_revision_id'] = $entity->getRevisionId();
+        $processed++;
+        $this->getLogger()->info(strtr('Successfully processed WFP Logcluster document @id to entity @entity_id.', [
+          '@id' => $id,
+          '@entity_id' => $entity->id(),
+        ]));
+      }
+      catch (\Exception $exception) {
+        $import_record['status'] = 'error';
+        $import_record['message'] = $exception->getMessage();
+        $import_record['attempts'] = ($import_record['attempts'] ?? 0) + 1;
+        $this->getLogger()->error(strtr('Unable to process WFP Logcluster document @id: @exception', [
+          '@id' => $id,
+          '@exception' => $exception->getMessage(),
+        ]));
+      }
+      finally {
+        $import_records[$import_record['imported_item_uuid']] = $import_record;
+      }
+    }
+
+    // Create or update the import records.
+    $this->saveImportRecords($import_records);
+
+    return $processed;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function processDocumentData(string $uuid, array $document): array {
+    $data = [];
+
+    // Source: Logistics Cluster.
+    $sources = [3594];
+
+    if (isset($document['organisations'])) {
+      foreach ($document['organisations'] ?? [] as $organisation) {
+        if ($organisation === 'WFP') {
+          continue;
+        }
+
+        if ($source = $this->getSourceByName($organisation)) {
+          $sources[] = $source;
+        }
+      }
+    }
+
+    // Document URL.
+    $url = $document['path'];
+
+    // Retrieve the title and clean it.
+    $title = $this->sanitizeText($document['title'] ?? '');
+
+    // The documents in the UNHCR API seldom have descriptions or good ones
+    // so we simply skip the body.
+    $body = '';
+
+    // Retrieve the publication date.
+    $published = strtotime($document['last_update'] ?? $document['date']);
+    $published = DateHelper::format($published, 'custom', 'c');
+
+    // Retrieve the document languages and default to English if none of the
+    // supported languages were found.
+    $languages = [];
+    if (isset($document['langcode'])) {
+      if (isset($this->languageMapping[$document['langcode']])) {
+        $languages[$document['langcode']] = $this->languageMapping[$document['langcode']];
+      }
+    }
+
+    if (empty($languages)) {
+      $languages['English'] = $this->languageMapping['English'];
```

Comment:
Shouldn't this use `en` instead of `English` since `$this->languageMapping` uses the language codes as keys not the full names?